pymongo/topology.py

```diff
@@ -128,15 +145,17 @@ def __init__(self, topology_settings):
         self._opened = False
         self._closed = False
         self._lock = _create_lock()
+        assert self._settings.condition_class is not None
         self._condition = self._settings.condition_class(self._lock)
-        self._servers = {}
-        self._pid = None
-        self._max_cluster_time = None
+        self._servers: Dict[_Address, Server] = {}
+        self._pid: Optional[int] = None
+        self._max_cluster_time: Optional[Mapping[str, Any]] = None
         self._session_pool = _ServerSessionPool()
 
         if self._publish_server or self._publish_tp:
+            weak: weakref.ReferenceType[Optional[queue.Queue]]
```

Comment:
Should be `ReferenceType[queue.Queue]` (ditch the optional).

Reply:
I see now, fixed!